Realtime docs: added limitations for row level changes

tchief opened the pull request:
## What kind of change does this PR introduce?

Docs updated with limitations for row level changes.

## What is the current behavior?

1. Only `eq` operator is available.
2. Only filtering by a single column is available.
3. UPDATE event is not fired if filtered by a changed column.

Example:
```
const mySubscription = supabase
  .from('forms:request_id=eq.200')
  .on('UPDATE', handleRecordUpdated)
  .subscribe()
```
Changing row `{ id: 1, request_id: 200 }` to `{ id: 1, request_id: 201 }` would not fire UPDATE event.
Which might be expected, similar way as DELETE event fired with `payload.old` property having removed row.

## What is the new behavior?

Added these notes to docs.

## Additional context

Related Issue: [#132](https://github.com/supabase/realtime/issues/132)
Related PR: [#133](https://github.com/supabase/realtime/pull/133)

tchief committed:
Realtime docs: added limitations for row level changes

tchief committed:
Realtime docs: updated limitations for row level changes

kiwicopple commented on web/spec/supabase.yml:
````diff
@@ -1451,7 +1451,7 @@ pages:
           ```
       - name: Listening to row level changes
-        description: You can listen to individual rows using the format `{table}:{col}=eq.{val}` - where `{col}` is the column name, and `{val}` is the value which you want to match.
+        description: You can listen to individual rows using the format `{table}:{col}=eq.{val}` - where `{col}` is the column name, and `{val}` is the value which you want to match. Only filtering by a single column and only using `eq`, `neq`, `lt`, `lte`, `gt`, `gte` operators is available now. Keep in mind, Supabase would not send an update event if `{col}` changed from `{val}` to `{val2}`.
````
```suggestion
        description: You can listen to individual rows using the format `{table}:{col}=eq.{val}` - where `{col}` is the column name, and `{val}` is the value which you want to match. Only filtering by a single column and only using `eq`, `neq`, `lt`, `lte`, `gt`, `gte` operators are available. Keep in mind, Supabase would not send an update event if `{col}` changed from `{val}` to `{val2}`.
```

Thanks for this @tchief - a munch needed update.

Can you clarify this sentence? 

```
Keep in mind, Supabase would not send an update event if `{col}` changed from `{val}` to `{val2}`.
```

This only pertains to the `eq` filter right?

tchief replied:
@kiwicopple This sentence is for `eq` filter, correct.

I expected it do be true for other filters, so just double-checked that. And it seems that `${table}:${column}=gte.${value}` doesn't produce realtime updates for filters others than `eq` at all.

I use it this way
```
supabase.from<T>(`${table}:${column}=gte.${value}`)
  .on('DELETE', handleDelete)
  .on('UPDATE', handleUpdate)
  .subscribe();
```

Maybe syntax is different for other operators. However, with `neq` I expect it to be the same, and it doesn't work.
Let me know, please, if there's another syntax to try other filters.

Otherwise I'll update PR and remove all the other operators list except `eq`.

kiwicopple replied:
Perhaps we can update it to 

```yaml
description: | 
    You can listen to individual rows using the format `{table}:{col}=eq.{val}` - where `{col}` is the column name, and `{val}` is the value which you want to match. 
    
    Only filtering by a single column and only using `eq`, `neq`, `lt`, `lte`, `gt`, `gte` operators are available. 
    When using the `eq` and `neq` filters, keep in mind that an event will not be sent if a column is updated to a value that no longer matches your filter.

```

tchief replied:
That's great!

One thing to mention, unfortunately Supabase doesn't indeed yet supports all the other operators for Realtime.
Here's the [link](https://github.com/supabase/realtime/issues/132#issuecomment-1089121668).

kiwicopple replied:
(so actually all of those filters should work)

tchief replied:
Should I enable WALRUS in my Supabase dashboard?
Using latest dependency `"@supabase/supabase-js": "^1.33.3"` still cannot make it send updates/deletes with 'gte'.

tchief replied:
Added code with steps to reproduce [here](https://github.com/tchief/supabase-js/commit/7abf9983cf19024e0937c8dc5baabda6379c5887), based on example from `supabase-js` repo.
Here's testing process

https://user-images.githubusercontent.com/729374/162426231-7a6071cd-aa82-4b51-8920-647739964ecc.mp4

tchief committed:
Update web/spec/supabase.yml

Co-authored-by: Copple <10214025+kiwicopple@users.noreply.github.com>

tchief committed:
Realtime: removed operators that don't supported yet

Only `eq` is available for Realtime. Other operators are available for WALRUS for now.
https://github.com/supabase/realtime/issues/132#issuecomment-1089121668

tchief committed:
Realtime: clarified description

tchief committed:
Realtime: removed duplicated description.

w3b6x9 commented:
I just wanted to clarify that WALRUS indeed works with additional filters but at the moment Realtime only supports `eq`. We are currently working on exposing the other filters in Realtime.

w3b6x9 commented:
@tchief thanks so much for this contribution. I'm going to close this for now, confirm the filters that work, and update the docs accordingly.
